src/components/Link/Link.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import classnames from 'classnames'
+
+type StyledLinkProps = {
+  variant?: 'external' | 'unstyled'
+  className?: string
+}
+
+type LinkProps = {
+  href: string
+  children: React.ReactNode
+} & React.PropsWithChildren<StyledLinkProps> &
+  JSX.IntrinsicElements['a']
+
+type AsCustomProps<T> = {
+  asCustom: keyof JSX.IntrinsicElements | React.FunctionComponent<T>
+} & StyledLinkProps &
+  React.PropsWithChildren<T>
+
+type PossibleLinkProps<T> = LinkProps | AsCustomProps<T>
+
+function isAsCustomProps<T>(
+  props: PossibleLinkProps<T>
+): props is AsCustomProps<T> {
+  return (props as AsCustomProps<T>).asCustom !== undefined
+}
+
+function linkClasses(
+  variant: StyledLinkProps['variant'],
+  className: StyledLinkProps['className']
+): string | undefined {
+  const unstyled = variant === 'unstyled'
+  const isExternalLink = variant === 'external'
+
+  return unstyled
+    ? className
+    : classnames(
+        'usa-link',
+        {
+          'usa-link--external': isExternalLink,
+        },
+        className
+      )
+}
+
+export function Link<FCProps = LinkProps>(
```

Comment:
I think we still need [overloads](https://mariusschulz.com/blog/function-overloads-in-typescript#version-4-function-overloads) to get accurate intellisense and type suggestions. What I notice is right now when the `asCustom` is not present  and `Link` has invalid props, it only warns about not including `asCustom`, maybe because it has the broadest types. It doesn't warn for anchor tag. 

Here's what an overload before line 46 could look (I think): 

```
// Overloads
export function Link<T>(asCustomProps: AsCustomProps<T>): React.ReactElement
export function Link(linkProps: LinkProps): React.ReactElement
```

 Lmk if you want to pair on this or if I'm off base.

Reply:
Thank you for this suggestion.  I'll have to look into this.  I think I will have time on Friday

Reply:
Thank you!  I think I've simplified the type signatures and added the overloads.  It seems to do what we expect in VSCode now.